src/factory.ts

```diff
@@ -23,15 +23,12 @@ export function createReducer<TState = any>(
   return function(state: any = initialState, action: Action) {
     const actionMeta = actions[action.type];
     if (actionMeta) {
-      const result = instance[actionMeta.fn](state, action);
+      const newStateInstance = Array.isArray(state) ? [...state] : { ...state };
+      let result = instance[actionMeta.fn](newStateInstance, action);
       if (result === undefined) {
-        if (Array.isArray(state)) {
-          return [...state];
-        } else {
-          return { ...state };
-        }
+        result = newStateInstance;
       }
-      state = result;
+      store = result;
```

Comment:
Why state was renamed to store?

Reply:
After building and testing, during PR preparation I made some changes and introduced wrong variable name by mistake. I corrected the variable name.